zerver/webhooks/gitlab/view.py

```diff
@@ -408,6 +410,77 @@ def get_release_event_body(payload: WildValue, include_title: bool) -> str:
     return body
 
 
+def get_emoji_event_transformed_type(payload: WildValue, type: str) -> str:
+    if type == "MergeRequest":
+        return "MR"
+    elif type == "DesignManagement::Design":
+        return "design"
+    elif type == "Note":
+        type = payload["note"]["noteable_type"].tame(check_string)
+        return get_emoji_event_transformed_type(payload, type)
+    return type.lower()
+
+
+def get_emoji_event_subtype_message(type: str) -> str:
+    return "a comment on " if type == "Note" else ""
+
+
+def get_emoji_event_url_id(payload: WildValue, type: str) -> tuple[str, str]:
+    if type == "design":
+        comment_url, _, design_name = parse_design_comment(
+            payload["note"], payload["project"]["web_url"].tame(check_string)
+        )
+        return comment_url, design_name
+
+    url = payload["object_attributes"]["awarded_on_url"].tame(check_string)
+    return url, url.split("/")[-1].split("#")[0]
+
+
+def get_emoji_event_topic_title(
+    payload: WildValue, awardable_type: str, type: str, use_merge_request_title: bool
+) -> str:
+    """
+    The payload format depends on the event target type (MR / issue / snippet).
+    Get topic from the corresponding payload section's title.
+    """
+    if type == "MR":
+        type = "merge_request" if use_merge_request_title else ""
+    elif type == "issue":
+        type = "issue" if awardable_type == "Note" else "work_item"
+    elif type == "snippet":
+        type = "project_snippet"
+
+    if type:
+        return payload[type]["title"].tame(check_string)
+
+    return ""
+
+
+def get_emoji_event_number_sign(type: str) -> str:
+    return "" if type == "design" else "#"
+
+
+def get_emoji_event_body(action: str, payload: WildValue, include_title: bool) -> str:
+    transformed_action = {"award": "added", "revoke": "removed"}.get(action, "reacted")
+    preposition = {"award": "to", "revoke": "from"}.get(action, "to")
+    emoji = payload["object_attributes"]
+    awardable_type = emoji["awardable_type"].tame(check_string)
+    transformed_type = get_emoji_event_transformed_type(payload, awardable_type)
+    url, id = get_emoji_event_url_id(payload, transformed_type)
+
+    return EMOJI_MESSAGE_TEMPLATE.format(
+        user_name=get_issue_user_name(payload),
+        action=transformed_action,
+        preposition=preposition,
+        type=transformed_type,
+        subtype=get_emoji_event_subtype_message(awardable_type),
+        number_sign=get_emoji_event_number_sign(transformed_type),
```

Comment:
Why are we doing these inline while doing `url`, `id` and others before? Can we do the one thing for all for consistency?